webapp/channels/src/components/post_view/gm_conversion_message/gm_conversion_message.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useRef} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+import {useDispatch, useSelector} from 'react-redux';
+
+import type {Post} from '@mattermost/types/posts';
+
+import {getMissingProfilesByIds} from 'mattermost-redux/actions/users';
+import {makeGetProfilesByIdsAndUsernames} from 'mattermost-redux/selectors/entities/users';
+
+import {renderUsername} from 'components/post_markdown/system_message_helpers';
+
+import type {GlobalState} from 'types/store';
+
+export type Props = {
+    post: Post;
+}
+function GMConversionMessage(props: Props): JSX.Element {
+    const convertedByUserId = props.post.props.convertedByUserId;
+    const gmMembersDuringConversionIDs = props.post.props.gmMembersDuringConversionIDs as string[];
+
+    const dispatch = useDispatch();
+    const intl = useIntl();
+
+    useEffect(() => {
+        dispatch(getMissingProfilesByIds(gmMembersDuringConversionIDs));
+    }, [props.post]);
+
+    const getProfilesByIdsAndUsernames = useRef(makeGetProfilesByIdsAndUsernames());
+    const userProfiles = useSelector(
+        (state: GlobalState) => getProfilesByIdsAndUsernames.current(
+            state,
+            {allUserIds: gmMembersDuringConversionIDs, allUsernames: []},
+        ),
+    );
+
+    const convertedByUserUsername = userProfiles.find((user) => user.id === convertedByUserId)!.username;
+    const gmMembersUsernames = userProfiles.map((user) => renderUsername(user.username));
```

Comment:
The maximum number of elements is low, so this should be fine. But we could consider doing a useMemo over `userProfiles`. 0/5.

Reply:
Noted for follow up PR.